lib/sisimai/bite/email/exchange2007.rb

```diff
@@ -0,0 +1,178 @@
+        :rhost  => %r/\AGenerating[ ]server:[ ]?(.*)/,
+        :rfc822 => %r/\AOriginal message headers:/,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      NDRSubject = {
+        :'SMTPSEND.DNS.NonExistentDomain'=> 'hostunknown',   # 554 5.4.4 SMTPSEND.DNS.NonExistentDomain
+        :'SMTPSEND.DNS.MxLoopback'       => 'networkerror',  # 554 5.4.4 SMTPSEND.DNS.MxLoopback
+        :'RESOLVER.ADR.BadPrimary'       => 'systemerror',   # 550 5.2.0 RESOLVER.ADR.BadPrimary
+        :'RESOLVER.ADR.RecipNotFound'    => 'userunknown',   # 550 5.1.1 RESOLVER.ADR.RecipNotFound
+        :'RESOLVER.ADR.ExRecipNotFound'  => 'userunknown',   # 550 5.1.1 RESOLVER.ADR.ExRecipNotFound
+        :'RESOLVER.ADR.RecipLimit'       => 'toomanyconn',   # 550 5.5.3 RESOLVER.ADR.RecipLimit
+        :'RESOLVER.ADR.InvalidInSmtp'    => 'systemerror',   # 550 5.1.0 RESOLVER.ADR.InvalidInSmtp
+        :'RESOLVER.ADR.Ambiguous'        => 'systemerror',   # 550 5.1.4 RESOLVER.ADR.Ambiguous, 420 4.2.0 RESOLVER.ADR.Ambiguous
+        :'RESOLVER.RST.AuthRequired'     => 'filtered',      # 550 5.7.1 RESOLVER.RST.AuthRequired
+        :'RESOLVER.RST.NotAuthorized'    => 'rejected',      # 550 5.7.1 RESOLVER.RST.NotAuthorized
+        :'RESOLVER.RST.RecipSizeLimit'   => 'mesgtoobig',    # 550 5.2.3 RESOLVER.RST.RecipSizeLimit
+        :'QUEUE.Expired'                 => 'expired',       # 550 4.4.7 QUEUE.Expired
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Microsoft Exchange Server 2007'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['Content-Language']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Microsoft Exchange Server 2007
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        return nil unless mhead['subject'] =~ Re0[:'subject']
+        return nil unless mhead['content-language']
+        return nil unless mhead['content-language'] =~ Re0[:'content-language']
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [84.88/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->